ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PointSchedule.hs

```diff
@@ -156,17 +168,17 @@ peerStates Peer {name, value = schedulePoints} =
 -- | Convert several @SinglePeer@ schedules to a common 'NodeState' schedule.
 --
 -- The resulting schedule contains all the peers. Items are sorted by time.
-peersStates :: PointSchedule blk -> [(Time, Peer (NodeState blk))]
+peersStates :: PointSchedule blk -> [(Time, PeerStateChange blk)]
 peersStates PointSchedule{psSchedule, psMinDuration} =
   case foldr (mergeOn fst) [] (peerStates <$> peersList psSchedule) of
-    [] -> []
+    [] -> if psMinDuration > Time 0 then [(psMinDuration, NoChange)] else []
     peers -> case last peers of
-      (t, p) | t < psMinDuration -> peers ++ [(psMinDuration, p)]
+      (t, _) | t < psMinDuration -> peers ++ [(psMinDuration, NoChange)]
       _                          -> peers
```

Comment:
This is the only place where `NoChange` is constructed. Is it necessary to introduce a new kind of tick for this?

`runSchedule` could just call `threadDelay` once more at the end if needed, which would be a more contained change.

Reply:
hm, and then perhaps the printing of the point schedule would need an extra row to remind the reader of the minimum duration.